core/src/main/java/org/jruby/RubyTime.java

```diff
@@ -1201,16 +1207,45 @@ public static IRubyObject at(ThreadContext context, IRubyObject recv, IRubyObjec
             millisecs = RubyNumeric.num2long(arg1) * 1000;
         }
 
+        String unit;
+        if (arg3 instanceof RubySymbol) {
+            unit = ((RubySymbol) arg3).toString();
+            throw context.runtime.newArgumentError("unexpected unit " + arg3);
```

Comment:
will always be raising whenever a unit Symbol arg is passed in
seems like the logic to validate supported units is missing here?

Reply:
@kares you were right, fixed it now!